lib/dw/dataset/addComputedColumns.js

```diff
@@ -6,33 +6,156 @@ import arrayMean from 'd3-array/src/mean';
 import arrayMedian from 'd3-array/src/median';
 import column from './column';
 import columnNameToVariable from '@datawrapper/shared/columnNameToVariable';
+import { Parser } from '../utils/parser';
 
-export default function(chart, dataset) {
-    const virtualColumns = chart.getMetadata('describe.computed-columns', {});
+export default function addComputedColumns(chart, dataset) {
+    const virtualColumns = chart.get('metadata.describe.computed-columns', {});
     const data = dataset.list();
     const columnNameToVar = {};
-    const columnAggregates = {};
+    const colAggregates = {};
+    const parser = new Parser();
 
     dataset.eachColumn(function(col) {
         if (col.isComputed) return;
         columnNameToVar[col.name()] = columnNameToVariable(col.name());
         if (col.type() === 'number') {
-            columnAggregates[col.name()] = {
-                min: arrayMin(col.values()),
-                max: arrayMax(col.values()),
-                sum: arraySum(col.values()),
-                mean: arrayMean(col.values()),
-                median: arrayMedian(col.values())
+            const [min, max] = col.range();
+            colAggregates[col.name()] = {
+                min,
+                max,
+                sum: sum(col.values()),
+                mean: mean(col.values()),
+                median: median(col.values())
             };
+        } else if (col.type() === 'date') {
+            const [min, max] = col.range();
+            colAggregates[col.name()] = { min, max };
         }
     });
 
-    _.each(virtualColumns, addComputedColumn);
+    // initialize meta objects for each computed column
+    const vNamesToVar = Object.keys(virtualColumns).reduce((acc, val, idx) => {
+        const key = columnNameToVariable(val);
+        return acc.set(key, {
+            name: val,
+            origIndex: idx,
+            key,
+            formula: virtualColumns[val],
+            visited: 0,
+            computed: false,
+            dependsOn: []
+        });
+    }, new Map());
+
+    // parse formulas to detect cross-column dependencies
+    _.each(virtualColumns, (formula, name) => {
```

Comment:
[suggestion]  
For a non-underscore approach you could do this:
```suggestion
    Object.entries(virtualColumns).forEach(([name, formula]) => {
```

I'm not sure which one I prefer though.

Reply:
I think in general it's a good idea to get rid of underscore dependencies. So I guess I like `Object.entries` more